handlers/spaces.go

```diff
@@ -153,5 +170,43 @@ func showSpace(w http.ResponseWriter, r *http.Request, folder *models.Folder, er
 		return
 	}
 
-	views.ShowSpace(c, space, userSpaces, folder, validationErrors).Render(r.Context(), w)
+	pagination := getPagination(r)
+	folders, err := getFolders(r, pagination)
+	if err != nil {
+		c.HandleError(w, r, err)
+		return
+	}
+
+	q, _ := pagination.Filter("q")
+	views.ShowSpace(c, space, folders, q.Value, userSpaces, folder, validationErrors).Render(r.Context(), w)
```

Comment:
Not sure if this is ok. Ignoring the "ok" result from `pagination.Filter()` because if it isn't, ok the default is still an empty string. The alternative would be a wormhole like:
```go
if !ok {
    q = ""
}
views.ShowSpace(...)
```
Another option would be to hand the full Pagination struct to the view and extract the `q`-value (and later also sort, offset and limit) there.

Reply:
So do we actually need an ok second return value for the `.Filter()` method then?

Reply:
Not right now, but we do if we want to know if the filter is present in future, or if "" is a meaningful value